assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -3682,94 +3723,159 @@ private String getContentReviewAcceptedFileTypesMessage() {
+            } else {
+                context.put("selectedCategories", state.getAttribute(NEW_ASSIGNMENT_CATEGORY));
+            }
+        } else {
+            String aTitle = assignment != null ? assignment.getTitle() : null;
+
+            HashMap<String, String> gAssignmentIdTitles = new HashMap<>();
+
+            HashMap<String, String> gradebookAssignmentsSelectedDisabled = new HashMap<>();
+            HashMap<String, String> gradebookAssignmentsLabel = new HashMap<>();
+
+            for (Assignment a : assignmentService.getAssignmentsForContext(contextString)) {
+                String gradebookItem = a.getProperties().get(PROP_ASSIGNMENT_ASSOCIATE_GRADEBOOK_ASSIGNMENT);
+                if (StringUtils.isNotBlank(gradebookItem)) {
+                    String associatedAssignmentTitles = "";
+                    if (gAssignmentIdTitles.containsKey(gradebookItem)) {
+                        // get the current associated assignment titles first
+                        associatedAssignmentTitles = gAssignmentIdTitles.get(gradebookItem) + ", ";
+                    }
 
-                // gradebook assignment label
-                String label = gAssignment.getName();
-                if (gAssignmentIdTitles.containsKey(gaId)) {
-                    label += " ( " + rb.getFormattedMessage("usedGradebookAssignment", new Object[]{gAssignmentIdTitles.get(gaId)}) + " )";
+                    // append the current assignment title
+                    associatedAssignmentTitles += a.getTitle();
+
+                    // put the current associated assignment titles back
+                    gAssignmentIdTitles.put(gradebookItem, associatedAssignmentTitles);
                 }
-                gradebookAssignmentsLabel.put(formattedText.escapeHtml(gaId), label);
             }
-        }
 
-        // Items sorted by name
-        gradebookAssignmentsLabel = gradebookAssignmentsLabel.entrySet().stream()
-                .sorted(Entry.comparingByValue(Comparator.nullsFirst(String.CASE_INSENSITIVE_ORDER)))
-                .collect(Collectors.toMap(Entry::getKey, Entry::getValue,
-                        (e1, e2) -> e1, LinkedHashMap::new));
+            // get all assignments in Gradebook
+            List gradebookAssignments = gradingService.getAssignments(contextString, contextString, SortType.SORT_BY_NONE);
 
-        context.put("gradebookAssignmentsSelectedDisabled", gradebookAssignmentsSelectedDisabled);
-        context.put("gradebookAssignmentsLabel", gradebookAssignmentsLabel);
+            // filtering out those from Samigo
+            for (Iterator i = gradebookAssignments.iterator(); i.hasNext(); ) {
+                org.sakaiproject.grading.api.Assignment gAssignment = (org.sakaiproject.grading.api.Assignment) i.next();
+                if (!gAssignment.getExternallyMaintained() || gAssignment.getExternallyMaintained() && gAssignment.getExternalAppName().equals(assignmentService.getToolId())) {
```

Comment:
Probably should simply ask the gardingService for only assignments with a specifc tool id vs getting all assignments and then attempting to filter them, this should be something the gradingService is responsible for

Reply:
This is actually the existing "site instance" code, we just left it as it is (and I think that's what should be done). It appears as modified because it's now inside an if block that checks if the feature is enabled, so we added some tabs.